### ISSUE TITLE
Remote execution of a SSIS Package

### ISSUE THREAD

**Issue by Seddryck** [Seddryck/NBi.SqlServer#1]
It's possible to remotely execute a SSIS package with the help of stored procedures such as:

```sql
-- Create the execution object
DECLARE @execution_id BIGINT
EXEC [SSISDB].[catalog].[create_execution]
@package_name = N'Package1.dtsx'
, @project_name = N'Project1'
, @folder_name = N'Folder1'
, @use32bitruntime = False
, @reference_id = NULL
, @execution_id = @execution_id OUTPUT

-- System parameters
EXEC [SSISDB].[catalog].[set_execution_parameter_value]
@execution_id
, @object_type = 50 -- System parameter
, @parameter_name = N'SYNCHRONIZED'
, @parameter_value = 1

-- Execute the package
EXEC [SSISDB].[catalog].[start_execution] @execution_id
```

NBi should support this way to execute a package.

**Comment by NeilRoper** [comment]
Thought I would share the workaround I implemented as a test. Nice to have features would be waiting on active packages to finish executing which I am doing here with a loop (not great) and status check. I added in the 'succeeded' and 'failed' results just for clarity but of course the assert could just be on the value 7.

<test name="Remote execution of a SSIS Package">
    <system-under-test>
      <resultSet>
        <query>
          <![CDATA[
            --Setup
            declare @execution_id BIGINT
            declare @status INT;
            declare @loop_count INT = 0;
            declare @loop_max INT = 9;
            declare @loop_delay_hh_mm_ss char(8) = '00:00:01';

            exec [SSISDB].[catalog].[create_execution]
              @package_name = N'Package1.dtsx'
            , @project_name = N'Project1'
            , @folder_name = N'Folder1'
            , @use32bitruntime = False
            , @reference_id = NULL
            , @execution_id = @execution_id output

            --Exercise
            exec [SSISDB].[catalog].[start_execution] @execution_id
            
            --Verify
            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id

            while (@status = 1 OR @status = 2 OR @status = 5)
	            and (@loop_max >= @loop_count)
	            begin
		            waitfor delay @loop_delay_hh_mm_ss
		            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id
		            set @loop_count = @loop_count + 1
	            end;

            --If we get here than the status is not active or the number of loops has been exceeded.
            if @status = 7 
	            select 'succeeded' as [result]
            else 
	            --either a genuine failure or the wait period was exceeded to assume failure.
	            --If the package did eventually succeed then adjust the loop delay wait period.
	            select 'failed' as [result]
  
            --Teardown
          ]]>
        </query>
      </resultSet>
    </system-under-test>
    <assert>
      <equalTo>
        <resultSet>
          <row>
            <cell>succeeded</cell>
          </row>
        </resultSet>
      </equalTo>
    </assert>
  </test>

**Comment by NeilRoper** [comment]
Thought I would share the workaround I implemented as a test. Nice to have features would be waiting on active packages to finish executing which I am doing here with a loop (not great) and status check. I added in the 'succeeded' and 'failed' results just for clarity but of course the assert could just be on the value 7.

<test name="Remote execution of SSIS package test">
    <system-under-test>
      <resultSet>
        <query>
          <![CDATA[
            --Setup
            declare @execution_id BIGINT
            declare @status INT;
            declare @loop_count INT = 0;
            declare @loop_max INT = 9;
            declare @loop_delay_hh_mm_ss char(8) = '00:00:01';

            exec [SSISDB].[catalog].[create_execution]
              @package_name = N'Package1.dtsx'
            , @project_name = N'Project1'
            , @folder_name = N'Folder1'
            , @use32bitruntime = False
            , @reference_id = NULL
            , @execution_id = @execution_id output

            --Exercise
            exec [SSISDB].[catalog].[start_execution] @execution_id
            
            --Verify
            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id

            while (@status = 1 OR @status = 2 OR @status = 5)
	            and (@loop_max >= @loop_count)
	            begin
		            waitfor delay @loop_delay_hh_mm_ss
		            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id
		            set @loop_count = @loop_count + 1
	            end;

            --If we get here than the status is not active or the number of loops has been exceeded.
            if @status = 7 
	            select 'succeeded' as [result]
            else 
	            --either a genuine failure or the wait period was exceeded to assume failure.
	            --If the package did eventually succeed then adjust the loop delay wait period.
	            select 'failed' as [result]
  
            --Teardown
          ]]>
        </query>
      </resultSet>
    </system-under-test>
    <assert>
      <equalTo>
        <resultSet>
          <row>
            <cell>succeeded</cell>
          </row>
        </resultSet>
      </equalTo>
    </assert>
  </test>

**Comment by NeilRoper** [comment]
Thought I would share the workaround I implemented as a test. Nice to have features would be waiting on active packages to finish executing which I am doing here with a loop (not great) and status check. I added in the 'succeeded' and 'failed' results just for clarity but of course the assert could just be on the value 7.

```
<test name="Remote execution of SSIS package test">
    <system-under-test>
      <resultSet>
        <query>
          <![CDATA[
            --Setup
            declare @execution_id BIGINT
            declare @status INT;
            declare @loop_count INT = 0;
            declare @loop_max INT = 9;
            declare @loop_delay_hh_mm_ss char(8) = '00:00:01';

            exec [SSISDB].[catalog].[create_execution]
              @package_name = N'Package1.dtsx'
            , @project_name = N'Project1'
            , @folder_name = N'Folder1'
            , @use32bitruntime = False
            , @reference_id = NULL
            , @execution_id = @execution_id output

            --Exercise
            exec [SSISDB].[catalog].[start_execution] @execution_id
            
            --Verify
            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id

            while (@status = 1 OR @status = 2 OR @status = 5)
	            and (@loop_max >= @loop_count)
	            begin
		            waitfor delay @loop_delay_hh_mm_ss
		            select @status = [status] from [SSISDB].[catalog].[executions] where execution_id = @execution_id
		            set @loop_count = @loop_count + 1
	            end;

            --If we get here than the status is not active or the number of loops has been exceeded.
            if @status = 7 
	            select 'succeeded' as [result]
            else 
	            --either a genuine failure or the wait period was exceeded to assume failure.
	            --If the package did eventually succeed then adjust the loop delay wait period.
	            select 'failed' as [result]
  
            --Teardown
          ]]>
        </query>
      </resultSet>
    </system-under-test>
    <assert>
      <equalTo>
        <resultSet>
          <row>
            <cell>succeeded</cell>
          </row>
        </resultSet>
      </equalTo>
    </assert>
  </test>
```
